api/src/main/java/io/minio/DestinationInfo.java

```diff
@@ -0,0 +1,73 @@
+package io.minio;
+
+
+import java.util.Map;
+
+
+/** DestinationInfo creates the DestinationObject
+ * used for Compose Object.
+*/
+public class DestinationInfo {
```

Comment:
I am not favour of this class in java supporting polymorphism.  `composeObject()` may have below versions
```java
composeObject(bucketName String, objectName String, ComposeSource []sources)
composeObject(bucketName String, objectName String, ServerSideEncryption sse, Map<String,String> userMetaData, ComposeSource []sources)
```